src/staging-heap.h

```diff
@@ -0,0 +1,161 @@
+#pragma once
+
+#include <slang-rhi.h>
+
+#include "core/common.h"
+#include "reference.h"
+
+#include "rhi-shared-fwd.h"
+
+#include <list>
+#include <unordered_map>
+
+namespace rhi {
+
+class StagingHeap: public RefObject
```

Comment:
looks like you didn't run formatting yet?
you can run `pre-commit run --all-files` to manually format code